exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetMetadataCache.java

```diff
@@ -446,10 +447,25 @@ public void testMetadataCacheAbsolutePaths() throws Exception {
     }
   }
 
+  @Test
```

Comment:
Tests for the version checks? A bit tricky because you have to create a file and "predict" the next version.

For this, you need a way to get the current version from your `Version` enum. Then do some simple logic to try:

* Add .1 to version number: 3.1 becomes 3.2.
* Move to next whole version number: 3.1 becomes 4.

Then, create a file with that version and check how this version of the code handles a future version of the file.

To be thorough, also create a file that contains a field that Jackson cannot deserialize. Be sure we properly handle a "hard" version incompatibility as discussed earlier.